src/hotspot/share/opto/vectornode.hpp

```diff
@@ -753,7 +761,15 @@ class StoreVectorNode : public StoreNode {
 
   virtual uint ideal_reg() const  { return Matcher::vector_ideal_reg(memory_size()); }
   virtual BasicType memory_type() const { return T_VOID; }
-  virtual int memory_size() const { return vect_type()->length_in_bytes(); }
+  virtual int memory_size() const {
+    if (in(MemNode::ValueIn)->bottom_type()->isa_vect() != NULL) {
+      return vect_type()->length_in_bytes();
```

Comment:
The same issue also exists when method `"vect_type()"` (line 757) is called directly, doesn't it?

Reply:
All codes that calles `vect_type()` in `StoreVectorNode` met the same issue. And I think it's better to make sure the `in(MemNode::ValueIn)->bottom_type()` is a `TypeVect`. We need to fix the issues that make it not the right type, doesn't it?

Reply:
Good catch!
Thanks @XiaohongGong for your review.

Yes, in theory, there may be the same bug when vect_type() is called since it's based on Type::is_vect().

However, by this reasoning, there might be the same bug in non-vector types such as Type::is_int()/Type::is_long()/Type::is_ptr()/... too.

Will do more investigation.
Thanks.